spec/requests/api/connect/v3/systems/products_controller_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+RSpec.describe Api::Connect::V3::Systems::ProductsController do
+  include_context 'auth header', :system, :login, :password
+  include_context 'version header', 3
+
+  let(:url) { connect_systems_products_url }
+  let(:headers) { auth_header.merge(version_header) }
+  let(:system) { FactoryGirl.create(:system) }
+  let(:product_with_repos) { FactoryGirl.create(:product, :with_repositories) }
+
+  describe '#activate' do
+    it_behaves_like 'products controller action' do
+      let(:verb) { 'post' }
+    end
+
+    context 'when product has repos' do
+      let(:payload) do
+        {
+          identifier: product_with_repos.identifier,
+          version: product_with_repos.version,
+          arch: product_with_repos.arch
+        }
+      end
+      let(:serialized_json) do
+        ActiveModelSerializers::SerializableResource.new(
+          product_with_repos.service,
+          serializer: ::V3::ServiceSerializer,
+          uri_options: { scheme: request.scheme, host: request.host, port: request.port },
+          status: status,
+          service_url: service_url(product_with_repos.service)
```

Comment:
We need to do some URL replacing, and `request` is not available in the models -- so kinda fishy.
But yeah, scheme, host and port are perhaps way too verbose.